code/controllers/_DynamicAreaLighting_TG.dm

```diff
@@ -209,15 +209,32 @@ atom/proc/AddLuminosity(delta_luminosity)
 //This slightly modifies human luminosity. Source of light do NOT stack.
 //When you drop a light source it should keep a running total of your actual luminosity and set it accordingly.
 mob/SetLuminosity(new_luminosity, trueLum)
-	if(!new_luminosity) //Our new addition is positive. Add it to our running total.
-		return..(0, trueLum)  //Set to ZERO.
+	//message_admins("MOB SET LUM DEBUG 1: luminosity_total: [luminosity_total] new_luminosity: [new_luminosity] length: [length(light_sources)]")
+	if( new_luminosity > 0 ) //If positive add a light source to the light source list.
+		light_sources.Add(new_luminosity)
+	else if ( new_luminosity < 0 ) //If negative, subtract the light source from the list instead.
+		light_sources.Remove(new_luminosity * -1)
+
+	if( luminosity_total > new_luminosity ) //Check to see if the new light source is more powerful than the current one; if so not, check for light loss
+		if(new_luminosity > -1) //We're not losing a light source; abort.
+			return
+		if(luminosity_total <= (new_luminosity * -1) ) //We're losing a light source of magnitude equal to or greater than our current one; recalc luminosity from the list of remaining sources
```

Comment:
This should only be true if `new_luminosity < 0` so why not nest it under it?

Reply:
Instead of having several separate checks, you could nest it in one check under another, following the logic of what's possible and avoiding multiple checks.
You can do a `switch(new_luminosity)`, for values zero, negative and positive.